feat: record all peer rps in grafana for simulations to calculate avg/min/max

dav1do opened the pull request:
We capture all peers average RPS over the simulation in `simulation_peer_requests_per_second` with an attribute of `peer`. We still have `simulation_min_peer_requests_per_second`. The motivation is being able to identify if changing something increases node writes, but reduces recon sync. We want a more clear picture of total writes in the scenario.

example includes the `ceramic-new-streams-benchmark` and `recon-event-sync` scenarios:
<img width="1510" alt="image" src="https://github.com/3box/keramik/assets/5317198/b65cf4cd-e08f-4c8f-a947-f027ceb2826a">

3benbox commented:
What I'd rather see is just the rps per peer using labels on a single metric.
We can calculate max/min/avg from a query.

It would look something like `simulation_requests_per_second{peer="ceramic-0"}`
or maybe more accurately. `simulation_results_requests_per_second{peer="ceramic-0"}` since these are a calculation.

dav1do commented:
> What I'd rather see is just the rps per peer using labels on a single metric. We can calculate max/min/avg from a query.
> 
> It would look something like `simulation_requests_per_second{peer="ceramic-0"}` or maybe more accurately. `simulation_results_requests_per_second{peer="ceramic-0"}` since these are a calculation.

Reworked to include `simulation_peer_requests_per_second` with a `peer` attribute (and rebased with main).

3benbox commented on runner/src/simulate.rs:
```diff
@@ -205,6 +205,111 @@ impl MetricsCollection for PromMetricCollector {
 }
 
+#[derive(Debug, Clone)]
+struct PeerRequestInfo {
```
Just a name comment, I feel like we should have Metric in the name or something to distinguish this from Peer Requesting Info 😵 

PeerRequestMetricInfo?

3benbox reviewed (approved):
Comment about naming.
Feature is amazing!
Someone else should review for code correctness
